src/libostree/ostree-repo-commit.c

```diff
@@ -429,6 +429,20 @@ add_size_index_to_metadata (OstreeRepo        *self,
   return g_variant_ref_sink (g_variant_builder_end (builder));
 }
 
+static gboolean
+throw_min_free_space_error (OstreeRepo  *self,
+                            guint64      bytes_required,
+                            GError     **error)
+{
+  g_autofree char *formatted_free = g_format_size (bytes_required);
+  if (self->min_free_space_percent > 0)
+    return glnx_throw (error, "min-free-space-percent '%u%%' would be exceeded, %s requested",
```

Comment:
Right. I can't think of a perfect error message here since these checks run while writing a single file. We don't know the space requested for writing future files so `%s requested` might give a impression that, `%s` is the size we need to complete the whole transaction (which isn't the case).